tests/unit/server/api/v1/test_datasets.py

```diff
@@ -5878,6 +5880,181 @@ async def test_search_dataset_records_with_response_status_filter(
         )
         assert response.status_code == 200
 
+    async def test_search_dataset_records_with_record_vector(
+        self, async_client: "AsyncClient", mock_search_engine: SearchEngine, owner: User, owner_auth_header: dict
+    ):
+        workspace = await WorkspaceFactory.create()
+        dataset, _, records, *_ = await self.create_dataset_with_user_responses(owner, workspace)
```

Comment:
We should start thinking about not using these kind of helpers anymore. In the future we should create isolated tests that are easy to change and to understand.